packages/components/pie-modal/src/index.ts

```diff
@@ -347,11 +356,29 @@ export class PieModal extends PieElement implements ModalProps {
         }
     }
 
-    private _getHeaderButtonVariant (): 'ghost-secondary' | 'ghost-inverse' {
-        const isInverted = this.backgroundColor === 'brand-06';
-        const variant = isInverted ? 'ghost-inverse' : 'ghost-secondary';
+    private _getHeaderButtonVariant (): 'ghost-secondary' | 'ghost-inverse' | 'secondary' {
+        const { imageSlotMode, backgroundColor } = this;
+
+        // Handle the combinations of image slot and background color
+        const hasImageSlot = Boolean(imageSlotMode);
+        const hasBackgroundColor = Boolean(backgroundColor) && backgroundColor !== 'default';
 
-        return variant;
+        // default case: image slot is not present
+        if (!hasImageSlot) {
+            // No image slot present, determine variant based on background color
+            const isInverted = backgroundColor === 'brand-06';
+            const variant = isInverted ? 'ghost-inverse' : 'ghost-secondary';
+            return variant;
+        }
+
+        // image slot is present
+        if (imageSlotMode === 'illustration') {
+            if (hasBackgroundColor) {
+                return 'secondary';
+            }
+            return 'ghost-secondary';
+        }
+        return 'secondary';
```

Comment:
I think this should be equivalent, what do you think about the readability?
```suggestion
        if (imageSlotMode === 'illustration' && !hasBackgroundColor) {
            return 'ghost-secondary';
        }
        return 'secondary';
```

Reply:
Still looks readable 👍